internal/pkg/kube/peer_authentication_reconciler.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"context"
+	"fmt"
+
+	securityv1beta1 "istio.io/api/security/v1beta1"
+	typev1beta1 "istio.io/api/type/v1beta1"
+	"istio.io/client-go/pkg/apis/security/v1beta1"
+	applyconfigurationv1 "istio.io/client-go/pkg/applyconfiguration/meta/v1"
+	applyv1beta "istio.io/client-go/pkg/applyconfiguration/security/v1beta1"
+	"istio.io/istio/pkg/kube"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift-service-mesh/federation/internal/pkg/xds"
+)
+
+var _ Reconciler = (*PeerAuthResourceReconciler)(nil)
+
+type PeerAuthResourceReconciler struct {
+	client    kube.Client
+	namespace string
+}
+
+func NewPeerAuthResourceReconciler(client kube.Client, namespace string) *PeerAuthResourceReconciler {
+	return &PeerAuthResourceReconciler{
+		client:    client,
+		namespace: namespace,
+	}
+}
+
+func (r *PeerAuthResourceReconciler) GetTypeUrl() string {
+	return xds.PeerAuthenticationTypeUrl
+}
+
+func (r *PeerAuthResourceReconciler) Reconcile(ctx context.Context) error {
+	pa := &v1beta1.PeerAuthentication{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fds-strict-mtls",
+			Namespace: r.namespace,
+			Labels:    map[string]string{"federation.istio-ecosystem.io/peer": "todo"},
+		},
+		Spec: securityv1beta1.PeerAuthentication{
+			Selector: &typev1beta1.WorkloadSelector{
+				MatchLabels: map[string]string{
+					"app.kubernetes.io/name": "federation-controller",
+				},
+			},
+			Mtls: &securityv1beta1.PeerAuthentication_MutualTLS{
+				Mode: securityv1beta1.PeerAuthentication_MutualTLS_STRICT,
+			},
+		},
+	}
```

Comment:
Is it worth adding this to the Istio `ConfigFactory` for consistency?

Reply:
Config factory exists only because we had MCP and K8S mode and the factory encapsulated common logic, but now it's no longer needed. @bartoszmajsak also noted that this is weird pattern. I guess we will move all of that logic to reconcilers or we will refactor that factory to functions which take required arguments and it want be part of a struct, so I didn't add anything new to that factory. WDYT?

Reply:
I quite like that there is a separation of concerns between defining/generating the configuration and applying the configuration. It's a bit strange at the moment because these configs are then converted to `applyConfigurations` (largely to reduce additional calls to kube-apiserver), but this will no longer be required when we move to controller-runtime.

Reply:
> I guess we will move all of that logic to reconcilers
This also seems like a good option, but I'm of the opinion we should remain consistent with the current pattern until when this refactor takes place (assuming it does).

Reply:
> @bartoszmajsak also noted that this is weird pattern.

I was only curious about the struct itself and naming as I would probably just go with simple funcs. I think we will keep generators in some sort of fashion, but I believe we can review/revamp it as we go.

Reply:
@eoinfennessy I will move it out of the reconciler in a PR with unit tests.